### PR TITLE
fix(ClearAll): always display clearAll button, disabled when no filter

### LINKED ISSUE #1530: ClearAll should be shown deactivated when not relevant, instead of hidden
See https://github.com/algolia/instantsearch.js/issues/1455

### PR DESCRIPTION
**Pull request by bobylito**
**Summary**

ClearAll is hidden when there are no filters. Following on #1455, it can be confusing to have part of the UI appear / disappear, making it disabled is more subtle and is a very common pattern (disabled buttons)

Fix #1530

### CODE REVIEW

**Comment by vvo** [comment]
Should we add a CSS disabled class and style it or use the disabled attribute? I have no preference, opening the discussion

File: packages/react-instantsearch/src/components/ClearAll.enzyme.test.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+describe('ClearAll', () => {
+  it('is disabled when there is no filters', () => {
+    const refine = jest.fn();
+    const wrapper = mount(
+      <ClearAll refine={refine} items={[]}/>
+    );
+
+    const btn = wrapper.find('button');
+    expect(refine.mock.calls.length).toBe(0);
+    btn.simulate('click');
```
Comment (vvo) [inline packages/react-instantsearch/src/components/ClearAll.enzyme.test.js]: BOOM TESTS!

**Comment by vvo** [comment]
As for tests, what about we only test the optimistic case with enzyme (there are items, we can click on the button) and then just use snapshots for other tests? (items, no items)

**Comment by bobylito** [comment]
Previously approved :)

**Pull request merged**